integrations/azure/azure_integration/ocean.py

```diff
@@ -47,29 +60,88 @@ async def resync_resource_groups(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     """
     Re-syncs resource groups, this is done separately because the resource groups api is different from the other apis
     """
+    resource_selector = typing.cast(
+        AzureSpecificKindSelector, get_current_resource_config().selector
+    )
+    async with DefaultAzureCredential() as credential:
+        async with SubscriptionClient(credential=credential) as subscription_client:
+            async for subscriptions_batch in batch_resources_iterator(
+                subscription_client.subscriptions.list,
+            ):
+                tasks = [
+                    resource_group_iterator(
+                        credential=credential,
+                        subscription_id=subscription["subscription_id"],
+                        api_version=resource_selector.api_version,
+                    )
+                    for subscription in subscriptions_batch
+                ]
+                combine = aiostream.stream.merge(tasks[0], *tasks[1:])
+                async with combine.stream() as streamer:
+                    async for resource_groups_batch in streamer:
+                        yield resource_groups_batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.SUBSCRIPTION)
+async def resync_subscriptions(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """
+    Re-syncs subscriptions, this is done separately because the subscriptions api is different from the other apis
+    """
     async with DefaultAzureCredential() as credential:
-        async with ResourceManagementClient(
-            credential=credential, subscription_id=get_integration_subscription_id()
-        ) as client:
-            async for resource_groups_batch in batch_resources_iterator(
-                client.resource_groups.list,
-                api_version=get_current_resource_config().selector.api_version,
+        async with SubscriptionClient(credential=credential) as client:
```

Comment:
combine with the one before

Reply:
the credentials are used in the code afterwards not only in the subscription client, skipping this suggestion for now